clang/lib/CIR/CodeGen/CIRRecordLayoutBuilder.cpp

```diff
@@ -151,6 +151,17 @@ struct CIRRecordLowering final {
                                            numberOfChars.getQuantity());
   }
 
+  mlir::Type getBitfieldStorageType(unsigned numBits) {
+    unsigned alignedBits = llvm::alignTo(numBits, astContext.getCharWidth());
+    if (builder.isCIRIntTypeSize(alignedBits)) {
+      return builder.getUIntNTy(alignedBits);
+    } else {
+      mlir::Type type = getCharType();
+      return mlir::cir::ArrayType::get(type.getContext(), type,
```

Comment:
Out of curiosity, is there a reason why you picked arrays versus picking structs for this?

Reply:
have no idea :) it was a first thought came into  my mind - once we don't have arbitrary sized integers, we may have arbitrary length arrays of bytes.  Well, maybe anonymous structure type will also work here.